alphagov/gsp-teams
# Fix and improve canary

## What

Thus far, canary has been released with the use of Flux/HelmRelease.
This has recently changed and Flux/HelmRelease is no longer no longer
part of the system.

New way of assuring CD, is with the use of Concourse.

Canary is still capable of testing a lot of the system unknowingly.
The introduced pipeline will:
- Build an Image - testing Concourse builds
- Push the Image to Registry - Testing Harbor and Kiam
- Notarise the image - Testing notary
- Apply the chart - Test the release

The one thing we're giving up is, testing Concourse's capability of
talking with external sources such as GitHub and pulling for changes.

This could be handy in the future, if we choose to lock down the mesh
network around concourse. The `src` resource would fail, which may be
enough of a test.

## How to review

- Sanity check
- Deploy the pipeline
- See it succeed
